File: persistence/src/main/java/com/example/diploma/persistence/dto/admin/CategoryDto.java
package com.example.diploma.persistence.dto.admin;

import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.experimental.FieldDefaults;

@Data
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class CategoryDto {

    Long id;
    String url;
    String name;
    CatalogDto catalogDto;

}

File: admin/src/main/frontend/src/app/dto/CategoryDto.ts
import {CatalogDto} from "./CatalogDto";

export class CategoryDto {

	id: number;
	name: string;
	url: string;
	catalogDto: CatalogDto;

	constructor(id?: number,
							name?: string,
							url?: string,
							catalogDto?: CatalogDto) {
		this.id = id;
		this.name = name;
		this.url = url;
		this.catalogDto = catalogDto;
	}
}